### ISSUE TITLE
archived on CRAN

### ISSUE THREAD

**Issue by sckott** [ropensci/grainchanger#6]
@laurajanegraham https://cloud.r-project.org/web/packages/grainchanger/ your pkg was archived on cran. anything we can do to help?

**Comment by laurajanegraham** [comment]
Thanks for this. The issue is that one of the tests is now failing, but it's not failing for me. It looks like it's a rounding issue, but I cannot work out what has changed. 

I think for now I might remove this test unless you have any ideas as to how to investigate?

**Comment by sckott** [comment]
Hmm, what I would do is try to replicate the issue on the platform that they said it was failing on. If the platform is different from your computer, you could try rhub `remotes::install_github("r-hub/rhub")`, `rhub::platforms()` will show you the different platforms available, then you could run check on your package with the failing platform. Its possible the rhub check will fail for a reason unrelated to your package, in which case, move on from that. 

let me know if i can help any other way, i.e.,  i can look at the test, etc.

**Comment by laurajanegraham** [comment]
For now I've removed the test, as the amount it was failing by was fairly insignificant and related to precision. 

I plan to rewrite the test soon, but am currently swamped. I've resubmitted without the test. I think it's something to do with the calculation of max and min, but I can't get to the bottom of it. Have you seen this elsewhere?

**Comment by sckott** [comment]
sounds good. i never saw the actual test that failed, if you can show me i'll have a look

**Comment by laurajanegraham** [comment]
Thanks, if you have time, I'd really appreciate that as I can't work it out! 

It's the commented out test in here: tests/testthat/test-winmove_agg.R

**Comment by sckott** [comment]
that test works on my mac. looking at the historical cran checks  https://cranchecks.info/pkgs/grainchanger/history?offset=10 seems like the failures were happening on all linux checks, but not windows or mac - i can't see an immediate reason why its failing

**Comment by laurajanegraham** [comment]
Yeah, it's a bit odd. I think it's got to be something to do with floating point numbers. Although it's working on my university linux machine... I'll leave it without that test for now and now get on with the task of working out why things that weren't failing CRAN checks before are now!!

**Comment by laurajanegraham** [comment]
Hi @sckott, 

I've finally got back to trying to get this up on CRAN again. I'm having an issue with r-hub where it's having difficulty installing dependencies. 

`#> ERROR: dependency ‘units’ is not available for package ‘sf’`
`#> ERROR: dependency ‘openssl’ is not available for package ‘httr’`
`#> ERROR: dependency ‘xml2’ is not available for package ‘spelling’`
`#> ERROR: dependency ‘httr’ is not available for package ‘gh’`
`#> ERROR: dependency ‘httr’ is not available for package ‘covr’`
`#> ERROR: dependency ‘gh’ is not available for package ‘usethis’`
`#> ERROR: dependencies ‘sf’, ‘usethis’ are not available for package ‘grainchanger’`

My question on this then, is how to I address this on R-hub? Ordinarily I would install the missing dependencies manually, so I'm going to try including them in "suggests" to see if that helps, but other than that I'm not sure. 

This is the platform: Ubuntu Linux 16.04 LTS, R-release, GCC

Thanks in advance!

**Comment by laurajanegraham** [comment]
Quick update - what I suggested above doesn't work, and now I have a new error on the windows build: 

`#> unable to load shared object 'C:/Users/USEROnDeRZaITG/R/Rcpp/libs/x64/Rcpp.dll'`

**Comment by sckott** [comment]
Not sure about the r-hub problems. @maelle Any ideas on how to solve the r-hub issue?

**Comment by maelle** [comment]
Could you please open an issue in the rhub issue tracker? Your issue seems different from https://github.com/r-hub/rhub/issues/369

**Comment by maelle** [comment]
Are the packages you mentioned above supposed to be direct dependencies of your package or dependencies of its dependencies?

**Comment by laurajanegraham** [comment]
Thanks both. I was starting to wonder if it was an r-hub issue rather than something I'm doing. 

So the packages above are dependencies of dependencies. Does that make a difference? 

I'll open an issue on the rhub issue tracker. Thanks for your help :)

**Comment by laurajanegraham** [comment]
Apologies, a quick follow up question - is there no point trying to submit to CRAN while it's failling this one?

**Comment by maelle** [comment]
> So the packages above are dependencies of dependencies. Does that make a difference?

Yes in the sense that you shouldn't add them to DESCRIPTION. It might be a R-hub problem.

I think I'd wait for R-hub checks or look for an alternative solution to get builds on other OS (Winbuilder for instance, although one should/can use it after R-hub checks)

**Comment by laurajanegraham** [comment]
I think I've only got the packages that are direct dependencies in DESCRIPTION - did a check of that yesterday. But `sf` and `usethis` are direct dependencies and it's struggling to install those due to their dependencies. Winbuilder is passing thankfully.

**Comment by mhesselbarth** [comment]
#9
